acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+            "cyclones_stitch_{}_{}_{}.dat".format(ref_name, ref_start_yr, ref_end_yr),
+        )
+        ref_cyclones_file = os.path.join(
+            reference_data_path,
+            "cyclones_hist_{}_{}_{}.nc".format(test_name, test_start_yr, test_end_yr),
+        )
+        ref_cyclones_hist = cdms2.open(ref_cyclones_file)("density", squeeze=1)
+        ref_aew_file = os.path.join(
+            reference_data_path,
+            "aew_hist_{}_{}_{}.nc".format(test_name, test_start_yr, test_end_yr),
+        )
+        ref_aew_hist = cdms2.open(ref_aew_file)("density", squeeze=1)
+        ref_data["metrics"] = generate_tc_metrics_from_te_stitch_file(ref_te_file)
+        ref_data["cyclone_density"] = ref_cyclones_hist
+        ref_data["aew_density"] = ref_aew_hist
+        ref_num_years = int(ref_end_yr) - int(ref_start_yr) + 1
+        ref_data["aew_num_years"] = ref_num_years  # type: ignore
+        ref_data["cyclone_num_years"] = ref_num_years  # type: ignore
+        parameter.ref_title = "{} ({}-{})".format(
+            parameter.ref_name, ref_start_yr, ref_end_yr
+        )
+    elif run_type == "model_vs_obs":
+        ref_data["metrics"] = generate_tc_metrics_from_obs_files(reference_data_path)
+        ref_cyclones_file = os.path.join(
+            reference_data_path, "cyclones_hist_IBTrACS_1979_2018.nc"
+        )
+        ref_cyclones_hist = cdms2.open(ref_cyclones_file)(
+            "density", lat=(-60, 60, "ccb"), squeeze=1
+        )
+        ref_aew_file = os.path.join(reference_data_path, "aew_hist_ERA5_2010_2014.nc")
+        ref_aew_hist = cdms2.open(ref_aew_file)(
+            "density", lat=(0, 35, "ccb"), lon=(180, 360, "ccb"), squeeze=1
+        )
+        ref_data["cyclone_density"] = ref_cyclones_hist
+        ref_data["cyclone_num_years"] = 40  # type: ignore
+        ref_data["aew_density"] = ref_aew_hist
+        ref_data["aew_num_years"] = 1  # type: ignore
+        parameter.ref_name = "Observation"
+        parameter.ref_title = "Observation"
+    else:
+        raise ValueError("Invalid run_type={}".format(run_type))
+
+    tc_analysis_plot.plot(test_data, ref_data, parameter, BASIN_DICT)
+
+    return parameter
+
+
+def generate_tc_metrics_from_te_stitch_file(te_stitch_file: str) -> Dict[str, Any]:
+    """Generates tropical cyclone metrics from TE stitch file.
+
+    :param te_stitch_file: TE stitch file path
+    :type te_stitch_file: str
+    :return: Tropical cyclone metrics
+    :rtype: Dict[str, Any]
+
+    # TODO: Add tests to cover this function
+    """
+    print("\nGenerating TC Metrics from TE Stitch Files")
+    print("============================================")
+    with open(te_stitch_file) as f:
+        lines = f.readlines()
+
+    # Calculate number of storms and max length
+    num_storms, max_len = _calc_num_storms_and_max_len(lines)
+    # Parse variables from TE stitch file
+    te_stitch_vars = _get_vars_from_te_stitch(lines, max_len, num_storms)
```

Comment:
TE stitch variables were originally independent variables and not a dict.

This means that you need to pass each variable as function arguments, rather than a single dict.